app/cmd/start.go

```diff
@@ -112,94 +130,234 @@ func start(c *cli.Context) (err error) {
+	proxyGRPCServer, proxyGRPCListener, err := setupProxyGRPCServer(logsDir, proxyServiceAddress, diskServiceAddress, spdkServiceAddress, tlsConfig, shutdownCh)
 	if err != nil {
 		return err
 	}
-	hc := health.NewHealthCheckServer(pm)
+	go func() {
+		if err := proxyGRPCServer.Serve(proxyGRPCListener); err != nil {
+			logrus.WithError(err).Error("Stopping proxy gRPC server")
+		}
+		// graceful shutdown before exit
+		close(shutdownCh)
+	}()
+	logrus.Infof("Instance Manager proxy gRPC server listening to %v", proxyServiceAddress)
 
-	rpcService, listenAt, err := util.NewServer(listen, tlsConfig,
-		grpc.KeepaliveEnforcementPolicy(keepalive.EnforcementPolicy{
-			MinTime:             10 * time.Second,
-			PermitWithoutStream: true,
-		}),
-	)
+	// Start process manager server
+	pm, pmGRPCServer, pmGRPCListener, err := setupProcessManagerGRPCServer(processPortRange, logsDir, processManagerServiceAddress, tlsConfig, shutdownCh)
 	if err != nil {
-		return errors.Wrap(err, "failed to setup process gRPC server")
+		return err
 	}
-
-	rpc.RegisterProcessManagerServiceServer(rpcService, pm)
-	healthpb.RegisterHealthServer(rpcService, hc)
-	reflection.Register(rpcService)
-
 	go func() {
-		if err := rpcService.Serve(listenAt); err != nil {
-			logrus.Errorf("Stopping process gRPC server due to %v", err)
+		if err := pmGRPCServer.Serve(pmGRPCListener); err != nil {
+			logrus.WithError(err).Error("Stopping process manager gRPC server")
 		}
 		// graceful shutdown before exit
 		cleanup(pm)
 		close(shutdownCh)
 	}()
-	logrus.Infof("Instance Manager process gRPC server listening to %v", listen)
+	logrus.Infof("Instance Manager process manager gRPC server listening to %v", listen)
+
+	// Start SPDK server
+	if spdkEnabled {
+		spdkGRPCServer, spdkGRPCListener, err := setupSPDKGRPCServer(spdkPortRange, spdkServiceAddress, shutdownCh)
+		if err != nil {
+			return err
+		}
+		go func() {
+			if err := spdkGRPCServer.Serve(spdkGRPCListener); err != nil {
+				logrus.WithError(err).Error("Stopping SPDK gRPC server")
+			}
+			// graceful shutdown before exit
+			close(shutdownCh)
+		}()
+		logrus.Infof("Instance Manager SPDK gRPC server listening to %v", spdkServiceAddress)
+	}
 
 	sigs := make(chan os.Signal, 1)
 	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
 	go func() {
 		sig := <-sigs
 		logrus.Infof("Instance Manager received %v to exit", sig)
-		rpcService.Stop()
+		pmGRPCServer.Stop()
 	}()
 
 	return <-shutdownCh
 }
 
-func getProxyAddress(listen string) (string, error) {
+func getServiceAddresses(listen string) (processManagerServiceAddress, proxyServiceAddress, diskServiceAddress, instanceServiceAddress, spdkerviceAddress string, err error) {
 	host, port, err := net.SplitHostPort(listen)
 	if err != nil {
-		return "", err
+		return "", "", "", "", "", err
 	}
 
 	intPort, err := strconv.Atoi(port)
 	if err != nil {
-		return "", err
+		return "", "", "", "", "", err
+	}
+
+	return net.JoinHostPort(host, strconv.Itoa(intPort)),
```

Comment:
Just make the port should be compatible with old IM like process manager and proxy.

Reply:
Sure.
8500: ProcessManagerService
8501: ProxyService
8502: DiskService
8503: InstanceService
8504: SpdkService